apps/src/applab/appStorage.js

```diff
@@ -244,6 +246,35 @@ var handleDeleteRecord = function(tableName, record, onComplete, onError) {
   onComplete(true);
 };
 
+var recordListener = new RecordListener();
+
+/**
+ * This is a partial implementation of onRecordEvent in the following ways:
+ * 1. it polls instead of properly listening for notifications when data changes;
+ * 2. it only issues callbacks when records are created (not updated or deleted);
+ * 3. it assumes that record ids are strictly increasing. This is not currently true but
```

Comment:
Oops, this comment was out of date. This implementation of `onRecordEvent` should be fully functional, modulo performance concerns. I've updated the comment to reflect this.